### PR TITLE
RavenDB-22275 Updated to .NET 8

### PR DESCRIPTION
**Pull request by arekpalinski**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-22275/Update-v5.4-to-.NET-8

### Additional description

```
1. Run script (from the main directory):
> .\updateFrameworkVersion.ps1 8.0.4

2. Run script (from the main directory):
> .\updateTargetFramework.ps1 net8.0

3. Github actions - *.yml files

a) update DOTNET_VERSION (DOTNET_VERSION: 7.0.408 -> DOTNET_VERSION: 8.0.204) b) update dotnet-version ('7.0' -> '8.0' / '7.0.x' -> '8.0.x')

4. Update Directory.Build.props

a) <LangVersion>11</LangVersion> -> <LangVersion>preview</LangVersion> b) <PropertyGroup Condition="'$(TargetFramework)' == 'net6.0' OR '$(TargetFramework)' == 'net7.0'"> -> <PropertyGroup Condition="'$(TargetFramework)' == 'net6.0' OR '$(TargetFramework)' == 'net7.0' OR '$(TargetFramework)' == 'net8.0'">

5. Update <RuntimeIdentifiers> in .csproj files

a) Micro.Benchmark.csproj: win7-x64;win8-x64;win10-x64;ubuntu.14.04-x64 -> win-x64;win-x86;linux-x64
b) Voron.Benchmark.csproj: win7-x64;win8-x64;win10-x64;ubuntu.14.04-x64 -> win-x64;win-x86;linux-x64
c) Raven.Server.csproj: win7-x64;win8-x64;win81-x64;win10-x64;win7-x86;win8-x86;win81-x86;win10-x86;ubuntu.14.04-x64;ubuntu.16.04-x64;ubuntu.18.04-x64 -> win-x64;win-x86;linux-x64;osx-x64
d) Raven.Debug.csproj: win7-x64;win8-x64;win81-x64;win10-x64;win7-x86;win8-x86;win81-x86;win10-x86;ubuntu.14.04-x64;ubuntu.16.04-x64;ubuntu.18.04-x64 -> win-x64;win-x86;linux-x64;osx-x64
d) Raven.Migrator.csproj: win7-x64;win8-x64;win81-x64;win10-x64;win7-x86;win8-x86;win81-x86;win10-x86;ubuntu.14.04-x64;ubuntu.16.04-x64;ubuntu.18.04-x64 -> win-x64;win-x86;linux-x64;osx-x64
e) Voron.Recovery.csproj: win7-x64;win8-x64;win10-x64;ubuntu.14.04-x64 -> win-x64;win-x86;linux-x64
f) rvn.csproj: win7-x64;win8-x64;win81-x64;win10-x64;win7-x86;win8-x86;win81-x86;win10-x86;ubuntu.14.04-x64;ubuntu.16.04-x64;ubuntu.18.04-x64 -> win-x64;win-x86;linux-x64

6. Update files in docker/ directory: 7.0 -> 8.0

7. Update debug.Dockerfile file: 7.0 -> 8.0

8. Update scripts/certificates/powershell/dev-create-authenticated-cluster-with-subdomains.ps1: net7.0 -> net8.0

9. Update scripts/certificates/powershell/dev-create-authenticated-cluster.ps1: net7.0 -> net8.0

10. Update scripts/package.ps1: net7.0 -> net8.0

11. Update src/CommonAssemblyInfo.cs

12. Update src/Raven.Client/Properties/VersionInfo.cs

13. Update Raven.Client.csproj and

a) Add net8.0 to <TargetFrameworks>
b) Add new 'this is required for the nuget build to properly include the right files' section

14. Update Sparrow.csproj and

a) Add net8.0 to <TargetFrameworks>

15. Update src/Raven.Studio/scripts/typings.js: net7.0 -> net8.0

16. Update test/EmbeddedTests/EmbeddedTestBase.cs: net7.0 -> net8.0

17. Update test/EmbeddedTests/TestDriver/TestDriverExampleTest.cs: net7.0 -> net8.0
```

### Type of change

- [ ] Bug fix
- [ ] Regression bug fix
- [ ] Optimization
- [ ] New feature

### How risky is the change?

- [ ] Low 
- [x] Moderate 
- [ ] High
- [ ] Not relevant

### Backward compatibility

- [ ] Non breaking change
- [ ] Ensured. Please explain how has it been implemented?
- [ ] Breaking change
- [x] Not relevant

### Is it platform specific issue?

- [ ] Yes. Please list the affected platforms.
- [x] No

### Documentation update

- [ ] This change requires a documentation update. Please mark the issue on YouTrack using `Documentation Required` tag.
- [x] No documentation update is needed 

### Testing by Contributor

- [ ] Tests have been added that prove the fix is effective or that the feature works
- [ ] Internal classes added to the test class (e.g. entity or index definition classes) have the lowest possible access modifier (preferable `private`) 
- [ ] It has been verified by manual testing

### Testing by RavenDB QA team

- [ ] This change requires a special QA testing due to possible performance or resources usage implications (CPU, memory, IO). Please mark the issue on YouTrack using `QA Required` tag.
- [x] No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- [ ] Yes. Please list the affected features/subsystems and provide appropriate explanation
- [ ] No

### UI work

- [ ] It requires further work in the Studio. Please mark the issue on YouTrack using `Studio Required` tag.
- [x] No UI work is needed

### CODE REVIEW

File: src/Raven.Client/Properties/VersionInfo.cs [hunk 1]
```diff
@@ -4,7 +4,7 @@
 using Raven.Client.Extensions;
 using Raven.Client.Properties;
 
-[assembly: RavenVersion(Build = "54", CommitHash = "a377982", Version = "5.4", FullVersion = "5.4.118-custom-54", ReleaseDateString = "2024-03-19")]
+[assembly: RavenVersion(Build = "54", CommitHash = "a377982", Version = "5.4", FullVersion = "5.4.100-custom-54", ReleaseDateString = "2024-03-19")]
```
Comment (ppekrol) [inline src/Raven.Client/Properties/VersionInfo.cs]: 200

File: src/Raven.Server/Raven.Server.csproj [hunk 2]
```diff
@@ -173,12 +173,13 @@
     <PackageReference Include="Lucene.Net" Version="3.0.54011" />
     <PackageReference Include="Lucene.Net.Contrib.Spatial.NTS" Version="3.0.54011" />
     <PackageReference Include="McMaster.Extensions.CommandLineUtils" Version="4.1.1" />
+	<PackageReference Include="Microsoft.Bcl.AsyncInterfaces" Version="8.0.0" />
```
Comment (ppekrol) [inline src/Raven.Server/Raven.Server.csproj]: formatting

File: src/Raven.Client/Raven.Client.csproj [hunk 3]
```diff
@@ -134,7 +134,7 @@
   </ItemGroup>
   <ItemGroup>
     <PackageReference Include="Lambda2Js.Signed" Version="3.1.4" />
-    <PackageReference Include="Microsoft.AspNetCore.JsonPatch" Version="7.0.18" />
+    <PackageReference Include="Microsoft.AspNetCore.JsonPatch" Version="8.0.4" />
```
Comment (ppekrol) [inline src/Raven.Client/Raven.Client.csproj:146]: in this file, at the top there is a section that adds Sparrow.dll, you need to add net8 there as well

File: src/Raven.Client/Properties/VersionInfo.cs (hunk 1 above)
Comment (arekpalinski) [reply src/Raven.Client/Properties/VersionInfo.cs]: done

File: src/Raven.Server/Raven.Server.csproj (hunk 2 above)
Comment (arekpalinski) [reply src/Raven.Server/Raven.Server.csproj]: done

File: src/Raven.Client/Raven.Client.csproj (hunk 3 above)
Comment (arekpalinski) [reply src/Raven.Client/Raven.Client.csproj:146]: done

**Pull request merged**